woocommerce/woocommerce
# [COT/HPOS] Fix trashing/untrashing of orders

### Changes proposed in this Pull Request:

This pull request fixes some small bugs:

1. Trashing an order when COT was authoritative and sync was enabled didn't update the update date of the COT record. This caused that the next time the order was read it was untrashed (per the change introduced in https://github.com/woocommerce/woocommerce/pull/34465).
2. Similarly, trashing an order when COT was authoritative and sync was enabled wasn't trashing the equivalent entry in the posts table.
3. When an order was untrashed the corresponding `_wp_trash_meta_status` and `_wp_trash_meta_time` meta records weren't deleted. Thus if the order was trashed and untrashed repeatedly new meta rows would be created every time that would end up piling up in the meta table.
4. The change introduced in https://github.com/woocommerce/woocommerce/pull/34465 was unnecessarily re-syncing the orders when the update dates in COT and in the posts table were identical.
5. It was possible to load the order details in Admin when the order was trashed via direct URL. Now this shows an error in the same way as when posts is authoritative.

Closes #35089.

### How to test the changes in this Pull Request:

1. With COT enabled and sync active trash an order.
2. Verify that you can't access the order via Admin (navigate to `/wp-admin/admin.php?page=wc-orders&id=<order id>&action=edit` and you'll get an error).
3. Verify that the corresponding entry in the posts table is also trashed.
4. Go to the trashed orders view, then to the all orders view, verify that the order doesn't get untrashed automatically.
5. Untrash the order and verify that the order has the proper status and that there aren't `_wp_trash_meta_status` entries in the `wp_wc_orders_meta` table for the order.
6. Verify that the sync in the opposite direction (posts to COT) still works for the trash-untrash cycle (note that in this case you need to manually trigger the sync).
7. Manually alter the entry in posts table for an order, load the order, and verify that the data in COT has been synced with the changes you introduced (aka verify that https://github.com/woocommerce/woocommerce/pull/34465 still works).

### Other information:


### FOR PR REVIEWER ONLY:
